base/logger_console.go

```diff
@@ -117,27 +117,32 @@ func (l *ConsoleLogger) shouldLog(ctx context.Context, logLevel LogLevel, logKey
 		return false
 	}
 
-	return shouldLogConsoleDatabase(ctx, logLevel, logKey) ||
-		shouldLog(l.LogLevel, l.LogKeyMask, logLevel, logKey)
+	// database overrides console if set
+	shouldLogDb, ok := shouldLogConsoleDatabase(ctx, logLevel, logKey)
+	if ok {
+		return shouldLogDb
+	}
+
+	return shouldLog(l.LogLevel, l.LogKeyMask, logLevel, logKey)
```

Comment:
a few other places in the code have nil checks for l.LogLevel, i'm wondering if its needed here

Reply:
`LogLevel` and `LogKey` types are safe to use as `nil` values - the code fairly quickly gets down to these two methods:

https://github.com/couchbase/sync_gateway/blob/f7b2ac82b93a040b8533bb2f3f69015eb421ad75/base/log_level.go#L50-L53

https://github.com/couchbase/sync_gateway/blob/f7b2ac82b93a040b8533bb2f3f69015eb421ad75/base/log_keys.go#L134-L137

Reply:
LGTM, will ask someone else for a second pass